extensions/amp-analytics/0.1/activity-impl.js

```diff
@@ -124,7 +124,9 @@ const ACTIVE_EVENT_TYPES = [
  * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampDoc
  */
 export function installActivityServiceForTesting(ampDoc) {
-  registerServiceBuilderForDoc(ampDoc, 'activity', Activity);
+  registerServiceBuilderForDoc(ampDoc,
+      'activity',
+      ampdoc => new Activity(ampdoc));
```

Comment:
I'm actually going to recommend we go in the other direction, purely because it's less coding when registering a Constructor:

```js
// Constructor
function register(ampDoc) {
  registerServiceBuilderForDoc(ampDoc, 'example', Example);
}

// Factory
function register(ampDoc) {
  registerServiceBuilderForDoc(ampDoc, 'factory', (ampdoc) => {
    // Do stuff
    return new Example(ampdoc, otherstuff);
  });
}
```

Now, we always instantiate with new (you can return any object instance, that's how polymorphic constructors work in JS).